pynsee/utils/requests_session.py

```diff
@@ -0,0 +1,360 @@
+        self,
+        api_url=None,
+        sdmx_url=None,
+        file_format="application/xml",
+        print_msg=True,
+        raise_if_not_ok=False,
+    ):
+        "Performs a query to insee, either through API or sdmx_url"
+        result = None
+
+        try:
+            if api_url and self._query_is_valid_sirene_call(api_url):
+                if not self.sirene_key:
+                    commands = "\n\ninit_conn(sirene_key='my_sirene_key')\n"
+                    msg = (
+                        "Sirene key is missing, please check your credentials "
+                        "on portail-api.insee.fr !\n"
+                        "Please do the following to use your "
+                        f"credentials: {commands}\n\n"
+                        "If your token still does not work, please try to "
+                        "clear the cache :\n "
+                        "from pynsee.utils import clear_all_cache;"
+                        " clear_all_cache()\n"
+                    )
+                    raise ValueError(msg)
+
+                self._set_sirene_key()
+
+            if api_url:
+                result = self._request_api_insee(
+                    api_url,
+                    file_format=file_format,
+                    raise_if_not_ok=raise_if_not_ok,
+                    print_msg=print_msg,
+                )
+
+        except ValueError as e:
+
+            if self._query_is_valid_sirene_call(api_url):
+                # Log the error as a SIRENE API call can not function without
+                # key
+                logger.critical(e)
+
+        except requests.exceptions.RequestException:
+            if sdmx_url:
+                if print_msg:
+                    logger.critical("SDMX web service used instead of API")
+                result = self._request_sdmx_insee(
+                    sdmx_url, raise_if_not_ok=raise_if_not_ok
+                )
+
+        return result
+
+    def _query_is_valid_sirene_call(self, url):
+        "check if an URL is a valid API call to SIRENE"
+        return re.match(".*api-sirene.*", url)
+
+    def _set_sirene_key(self):
+
+        sirene_key = _get_credentials().get("sirene_key", None)
+        if not sirene_key:
+            raise ValueError
+        self.headers["X-INSEE-Api-Key-Integration"] = sirene_key
+
+    def _request_sdmx_insee(self, url, raise_if_not_ok=True):
+        results = self.get(url, verify=False)
+        if raise_if_not_ok and not results.ok:
+            raise requests.exceptions.RequestException(
+                results.text + "\n" + url, response=results
+            )
+        return results
+
+    def _request_api_insee(
+        self,
+        url,
+        file_format="application/xml",
+        raise_if_not_ok=False,
+        print_msg=True,
+    ):
+
+        headers = {"Accept": file_format}
+
+        try:
+            results = self.get(
+                url,
+                headers=headers,
+                verify=False,
```

Comment:
(why) do we need ``verify=False``

Reply:
Mostly not to trigger an error in case of certificate expiration. That's a trouble I have also encountered here and there, so I choose not to remove it.